improve github actions

SchSeba opened the pull request:
add cancel on progress and also run the virtual cluster tests only if all the light lanes like build and unit tests pass to save resources

bn222 commented:
Can you change the commit message to:

```
Speed up GitHub action runs and save resources

- Require short running actions (golangci, test, build) before starting slower virtual-k8s-cluster and only run virtual-ocp when virtual-k8s-cluster succeed
- Allow cancellation

These two changes together reduce the amount of resources used

```

bn222 commented:
LGTM

SchSeba commented:
/hold 
this doesn't run the virtual cluster now ;(

SchSeba commented:
I moved the virtual cluster to the same file as the build and unit test so we can have a dependency.
I also changed the name from Go to Test SR-IOV Operator to better represent the flow

SchSeba commented:
/hold cancel

zeeke reviewed (approved):
LGTM

adrianchiris commented on hack/run-e2e-conformance-virtual-cluster.sh:
```diff
@@ -130,18 +130,67 @@ kcli ssh $cluster_name-ctlplane-0 << EOF
 echo '$insecure_registry' > /etc/containers/registries.conf.d/003-internal.conf
 systemctl restart crio
+
+echo '[connection]
```
nit: define function and call it for each node ?

SchSeba replied:
done
